enhancements/machine-config/on-cluster-layering.md

```diff
@@ -0,0 +1,1137 @@
+map of ConfigMap name -> bootc annotations and store this in an additional
+ConfigMap. This would also necessitate the need for a binary such as Ignition
+live-apply to consume this information from within the build context to write
+those files into the appropriate places.
+
+This could look like this instead:
+
+```mermaid
+flowchart TB
+  Start("Start")
+  subgraph Prepare [Prepare For Build]
+    direction TB
+    RenderContainerfile("Render Containerfile")
+    PrepMachineConfig("Convert into bootc ConfigMaps")
+    PrepIndexConfigMap("Prepare bootc index ConfigMap")
+    ConfigMaps("Create ephemeral ConfigMaps")
+  
+    RenderContainerfile --> ConfigMaps
+    PrepMachineConfig --> PrepIndexConfigMap --> ConfigMaps
+  end
+  
+  subgraph ImageBuilder [Image Builder Context]
+    direction TB
+    ParseIndexConfigMap("Parse bootc index ConfigMap")
+    ApplyConfigMapsToImage("Apply bootc ConfigMaps \n to image build")
+    RunAdminProvidedContent("Run admin-provided \n Containerfile content")
+    ParseIndexConfigMap-->ApplyConfigMapsToImage-->RunAdminProvidedContent
+  end
+
+  Stop("Stop")
+
+  Start --> Prepare
+  ConfigMaps --> |Mounted as volumes into image build context| ImageBuilder
+  IsSuccess{"Is Build Successful?"}
+
+  ImageBuilder --> IsSuccess
+  IsSuccess --> | Yes | DeleteConfigMaps("Delete ConfigMaps")
+  IsSuccess --> | No | LeaveConfigMaps("Leave ConfigMaps \n for inspection")
+  DeleteConfigMaps --> Stop
+  LeaveConfigMaps --> Stop
+```
+
+To be clear, these ConfigMaps would be created and managed by the
+BuildController process. A cluster admin would never be expected to create or
+maintain these. This may also not be required if we were to adopt the [Custom OS
+Content Only](#####Custom-OS-Content-Only) idea.
+
+#### Build Pods
+
+Because we cannot rely upon OpenShift Image Builds to be present, there is
+the need for us to maintain our own customized build pod config. Doing this
+will allow OCP to have maximum control over the build process. Nevertheless,
+there are still some caveats that should be explicitly called out:
+
+- At the time of this writing, OpenShift Image Builder requires
+  privileged pods to perform its builds. Ideally, the custom build pod
+  solution should not have this requirement. Early experimentation
+  showed that Buildah can be used with a service account with the `anyuid`
+  ClusterRoleBinding applied without requiring elevated privileges. This
+  was done with an off-the-shelf Buildah image
+  (`quay.io/buildah/stable:latest`) and it was verified to work using the
+  lifecycled approach discussed below.
+
+- One of the promises that OpenShift makes is that all of the software
+  that is included in a given OpenShift release is tested to work
+  together, with no version change until an OpenShift upgrade is
+  performed. This same promise should extend to the image builder. To
+  implement this, Buildah is installed into the MCO image via the ART RPM
```

Comment:
So Buildah is already part of the OCP payload? We aren't adding a new dependency here?